tests/integration/operatorhub/cmd_service_test.go

```diff
@@ -449,7 +449,9 @@ spec:
 			stdOut = helper.CmdShouldFail("odo", "link", "EtcdCluster/example")
 			Expect(stdOut).To(ContainSubstring("already linked with the service"))
 
-			stdOut = helper.CmdShouldPass("odo", "unlink", "EtcdCluster/example")
+			// Running unlink command with timeout because at times unlink happens (which deletes underlying Kubernetes
+			// Secret) too fast and (seems to) mess the Pod created as a part of earlier odo push
+			stdOut = helper.CmdShouldRunWithTimeout(1*time.Minute, "odo", "unlink", "EtcdCluster/example")
```

Comment:
Looking at [your comment here](https://github.com/openshift/odo/issues/4301#issuecomment-776664917), wouldn't it make more sense to wait for the pod to be in a running state before this un-linking, instead of adding a static wait period?

Reply:
I agree with @valaparthvi that instead of introducing static sleep, we should properly wait for whatever is needed to make the operation successful. 

As I understand this is essentially a race condition. Should this be solved inside odo code? To me, it seems that we are fixing this in the test, but the actual problem is how odo handles this situation. 


Did you really mean to use `CmdShouldRunWithTimeout` ? This function just makes sure that the command is killed after a given timeout. It is not introducing any sleep before executing the command.

Reply:
Sure it kills the command but it also returns the stdout. And that's what we're interested in along with a sleep. However, I agree with what @valaparthvi has mentioned and am trying to change accordingly.

I think the race condition is happening because link and unlink operations are being executed fast in CI environment. This problem has not surfaced yet when someone does things manually. So I'm not sure if this needs to be fixed in the odo code.

Reply:
@valaparthvi @kadel I've made the requested changes. Can one of you PTAL?